components/RuntimeToProject.coffee

```diff
@@ -1,6 +1,34 @@
 noflo = require 'noflo'
 uuid = require 'uuid'
 
+fetchSources = (components, runtime, sources, callback) ->
+  return callback null, sources unless components.length
+  handleMessage = (msg) ->
+    if msg.command is 'error'
+      callback msg.payload
+      return
+    if msg.command is 'source'
+      sources.push msg.payload
+      fetchSources components, runtime, sources, callback
+      return
+    # We got unrelated message, subscribe again
+    runtime.once 'component', handleMessage
+  runtime.once 'component', handleMessage
+  component = components.shift()
+  runtime.sendComponent 'getsource',
+    name: component
+
+fetchFromLibrary = (runtime, callback) ->
+  return callback null, [] unless runtime.isConnected()
+  return callback null, [] unless runtime.canDo 'component:getsource'
+  return callback null, [] unless runtime.definition?.namespace
+  return callback null, [] unless runtime.definition.components
+  components = Object.keys(runtime.definition.components).filter (componentName) ->
```

Comment:
This code would be clearer if the filter function was moved out and given a name. `componentBelongsToProject` or similar